zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +96,174 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
+
+            //skip file type if its not supported
+            if (fileType == null) {
+                String msg = Messages.format("file.type.typeNotSupported", fileType);
+                consoleInteractor.printfln(DisplayMode.Warning, msg);
+                continue;
+            }
+
+            //throw error if file type is input more than once
+            if (filteredFileTypes.containsKey(inputFileType)) {
+                String msg =
+                        Messages.format("file.type.duplicateFileType",
+                                fileType);
+                log.error(msg);
+                throw new RuntimeException(msg);
+            }
+
+            Set<String> filteredExtensions = new HashSet<>(extensions);
+
+            validateSourceExtension(extensions, filteredExtensions,
+                    inputFileType);
+
+            /**
+             * Use the extensions from typeWithExtension input if exists,
+             * otherwise, use the extensions from server.
+             */
+            filteredExtensions =
+                filteredExtensions.isEmpty() ? inputFileType.getSourceExtensions()
+                            : filteredExtensions;
+
+            //throw error if same file extension found in multiple file type
+            for (DocumentType filteredDocType: filteredFileTypes.keySet()) {
+                for (String filteredExtension: filteredDocType.getSourceExtensions()) {
+                    if (filteredExtensions.contains(filteredExtension)) {
+                        String msg =
+                                Messages.format(
+                                        "file.type.conflictExtension",
+                                        fileType, filteredDocType.name(),
+                                        filteredExtension);
+                        log.error(msg);
+                        throw new RuntimeException(msg);
+                    }
+                }
+            }
+            filteredFileTypes.put(inputFileType, filteredExtensions);
+        }
+        return filteredFileTypes;
+    }
+
+    /**
+     * throw exception if user input extension if not part of DocumentType supported source extension
+     * @param extensions
+     * @param filteredExtensions
+     * @param inputFileType
+     */
+    private void validateSourceExtension(List<String> extensions,
+            Set<String> filteredExtensions, DocumentType inputFileType) {
+        if (!extensions.isEmpty()) {
+            for (String extension: extensions) {
+                if (!inputFileType.getSourceExtensions()
+                    .contains(extension)) {
+                    String msg = Messages.format(
+                        "file.type.notSupportedExtension", extension,
+                        inputFileType.name());
+                    consoleInteractor.printfln(DisplayMode.Warning, msg);
+                    filteredExtensions.remove(extension);
```

Comment:
- [ ] We should accept custom extensions, as long as the user specified a file type so we know what type to send the file as.